napari/_qt/dialogs/_tests/test_preferences_dialog.py

```diff
@@ -280,3 +283,70 @@ def test_preferences_dialog_restore(qtbot, pref, monkeypatch):
             ).text()
         )
     ) == KeyBinding.from_str('Ctrl')
+
+
+@skip_local_focus
+@skip_on_mac_ci
+@pytest.mark.key_bindings
+@pytest.mark.parametrize(
+    'confirm_key',
+    ['enter', 'return', 'tab', 'capslock'],
+)
+def test_preferences_dialog_not_dismissed_by_keybind_confirm(
+    qtbot, pref, confirm_key
+):
+    """This test ensures that when confirming a keybinding change, the dialog is not dismissed.
+
+    Notes:
+        * Skipped on macOS CI due to accessibility permissions not being
+          settable on macOS GitHub Actions runners.
+        * For this test to pass locally, you need to give the Terminal/iTerm/VSCode
+          application accessibility permissions:
+              `System Settings > Privacy & Security > Accessibility`
+
+        See https://github.com/asweigart/pyautogui/issues/247 and
+        https://github.com/asweigart/pyautogui/issues/247#issuecomment-437668855
+    """
+    shortcut_widget = (
+        pref._stack.widget(3).widget().widget.widgets['shortcuts']
+    )
+    pref._stack.setCurrentIndex(3)
+    # ensure the dialog is showing
+    pref.show()
+    assert pref.isVisible()
```

Comment:
maybe this should be a `qtbot.waitUntil`? It doesn't fail now but I've definitely had trouble before with a `show` followed immediately by an assert...

Reply:
Good idea -- I think I tried that and got an error. I will try again.

Reply:
Done, lets see if CI passes.